api/all/src/main/java/io/opentelemetry/api/trace/ImmutableSpanContext.java

```diff
@@ -14,21 +14,30 @@
 abstract class ImmutableSpanContext implements SpanContext {
 
   static final SpanContext INVALID =
-      create(
+      createInternal(
           TraceId.getInvalid(),
           SpanId.getInvalid(),
           TraceFlags.getDefault(),
           TraceState.getDefault(),
           /* remote= */ false);
 
+  private static AutoValue_ImmutableSpanContext createInternal(
+      String traceId, String spanId, TraceFlags traceFlags, TraceState traceState, boolean remote) {
+    return new AutoValue_ImmutableSpanContext(
+        traceId, spanId, traceFlags, traceState, /* remote$= */ remote);
+  }
+
   static SpanContext create(
       String traceIdHex,
       String spanIdHex,
       TraceFlags traceFlags,
       TraceState traceState,
       boolean remote) {
-    return new AutoValue_ImmutableSpanContext(
-        traceIdHex, spanIdHex, traceFlags, traceState, remote);
+    if (SpanId.isValid(spanIdHex) && TraceId.isValid(traceIdHex)) {
+      return createInternal(traceIdHex, spanIdHex, traceFlags, traceState, remote);
+    }
+    return createInternal(
```

Comment:
Should we return the INVALID SC?

Reply:
The reason is that it may seem strange to propagate flags and state but not one of the IDs if the other one is invalid

Reply:
So we either drop everything or we keep everything that is valid. I think the current implementation is in the middle and not ideal

Reply:
There were tests depending on this behavior. I'll need to figure out what they were trying to accomplish.

Reply:
I think I know why. There are 2 cases:
1. is invalid hex string -  in this case I think we should return SpanContext.getInvalid()
2. is equal with TraceIs.getInvalid() - this is used in tests I guess, probably we should preserve all values.

Not sure if that is the behavior we want, but this is a guess for the problem.

Reply:
yes...this will take a bit of research that we can create a follow-up issue to deal with, I think.

Reply:
Added #2751